dspace-server-webapp/src/main/java/org/dspace/app/rest/ItemUploadController.java

```diff
@@ -0,0 +1,155 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@RestController
+@RequestMapping("/api/core/items/{uuid}")
+public class ItemUploadController {
+
+    private static final Logger log = LogManager.getLogger();
+
+    @Autowired
+    protected Utils utils;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private BitstreamService bitstreamService;
+
+    @Autowired
+    private BitstreamConverter bitstreamConverter;
+
+    @Autowired
+    private MetadataConverter metadataConverter;
+
+    @Autowired
+    private BitstreamFormatService bitstreamFormatService;
+
+    @RequestMapping(method = RequestMethod.POST, value = "/bitstreams", headers = "content-type=multipart/form-data")
+    @PreAuthorize("hasPermission(#uuid, 'ITEM', 'WRITE') && hasPermission(#uuid, 'ITEM', 'ADD')")
+    public BitstreamResource uploadBitstream(HttpServletRequest request, @PathVariable UUID uuid,
+                                             @RequestParam("file") MultipartFile uploadfile,
+                                             @RequestParam(value = "properties", required = false) String properties) {
+
+        Context context = ContextUtil.obtainContext(request);
+        Item item = null;
+        Bitstream bitstream = null;
+        try {
+            item = itemService.find(context, uuid);
+        } catch (SQLException e) {
+            log.error("Something went wrong trying to find the Item with uuid: " + uuid, e);
+        }
+        if (item == null) {
+            throw new ResourceNotFoundException("The given uuid did not resolve to an Item on the server: " + uuid);
+        }
+        InputStream fileInputStream = null;
+        try {
+            fileInputStream = uploadfile.getInputStream();
+        } catch (IOException e) {
+            log.error("Something went wrong when trying to read the inputstream from the given file in the request");
+            throw new UnprocessableEntityException("The InputStream from the file couldn't be read");
+        }
+        try {
+            bitstream = processBitstreamCreation(context, item, fileInputStream, properties,
+                                                 uploadfile.getOriginalFilename());
+            itemService.update(context, item);
+            context.commit();
+        } catch (AuthorizeException | IOException | SQLException e) {
+            log.error(
+                "Something went wrong with trying to create the single bitstream for file with filename: " + uploadfile
+                    .getOriginalFilename()
+                    + " for item with uuid: " + uuid + " and possible properties: " + properties);
+
+        }
+        return new BitstreamResource(bitstreamConverter.fromModel(bitstream), utils);
+    }
+
+    private Bitstream processBitstreamCreation(Context context, Item item, InputStream fileInputStream,
```

Comment:
As this is a large method, I'd recommend JavaDocs here.

Reply:
I wouldn't know what to explain for this method apart from the fact that it creates the bitstream based on the given parameters
Is there perhaps one of the parts of the implementation not clear, or exactly what needs more detail

Reply:
"Creates the bitstream" is exactly the sort of explanation that I would appreciate, when I see the name elsewhere in the code and hover on it hoping to be told its function, rather than having to go to it and read the code.  Especially since "process" is rather vague and makes me wonder what, specifically, is happening there.

Generally, method comments are useful when I'm studying the method, but method doc comments are quite a bit more useful when I'm studying a *reference to* the method.  Reading the doco. in a popup is much less an interruption than wandering off to find the method and then trying to recall where I came from and what I was thinking then.

If it is a large method that's difficult to understand, maybe it should be a tree of smaller methods?

Reply:
A one line comment that says "create the bitstream based on the given parameters" is good enough here.  I didn't know what "process bitstream creation" meant at a glance...is it actually creating the bitstream? Or is it just processing the inputs of a creation request to send to another method?  A single line comment can make it clear what the method is doing without having to read all the code.  Or, you could rename the method to be "createBitstream" which would also be clearer.

Reply:
The JavaDocs have been included